subsys/bluetooth/mesh/adv_ext.c

```diff
@@ -343,8 +343,9 @@ static void adv_sent(struct bt_le_ext_adv *instance,
 	BT_DBG("Advertising stopped after %u ms", (uint32_t)duration);
 
 	atomic_clear_bit(adv->flags, ADV_FLAG_ACTIVE);
+	atomic_clear_bit(adv->flags, ADV_FLAG_PROXY);
 
-	if (!atomic_test_and_clear_bit(adv->flags, ADV_FLAG_PROXY)) {
+	if (adv->buf) {
 		net_buf_unref(adv->buf);
```

Comment:
Perhaps we could clear adv->buf here now too, to ensure there are no regressions for this check in the future?

Reply:
Logically, calling here is already an error